app/routes.py

```diff
@@ -0,0 +1,250 @@
+            return make_response("", 200)
+    
+    elif request.method == "PUT":
+        required_attributes = ["title", "release_date", "total_inventory"]
+        request_body = validate_data(request.get_json(), required_attributes)
+        video.title = request_body["title"]
+        video.total_inventory = request_body["total_inventory"]
+        video.release_date = request_body["release_date"]
+        db.session.commit()
+        return make_response(jsonify(video.to_dict()), 200)
+
+    else:
+        return make_response(jsonify({"message": "http request method not permitted"}), 400)
+
+@videos_bp.route('/<video_id>/rentals', methods=["GET"])
+def get_rentals_for_video(video_id):
+    """Defined an endpoint for getting all rentals associated with a video"""
+    video = is_class_type_at_id_found(video_id, Video, "Video")
+    rentals = video.video_rentals() 
+    return make_response(jsonify(rentals), 200)
+
+# Customer Routes
+@customers_bp.route("", methods=["GET", "POST"])
+def handle_customers():
+    """Defined an endpoint to handle all customer http request methods"""
+    if request.method == "GET":
+        query_sort = request.args.get("sort")
+        query_limit = request.args.get("n")
+        query_page = request.args.get("p")
+
+        customers = handle_query_params(query_sort, query_page, query_limit, Customer, Customer.name)
+
+        customers_list = []
+        for customer in customers:
+            customers_list.append(customer.to_dict())
+        return make_response(jsonify(customers_list), 200)
+
+    elif request.method == "POST":
+        required_attributes = ["postal_code", "name", "phone"]
+        request_body = validate_data(request.get_json(), required_attributes)
+        new_customer = Customer(
+            name = request_body["name"],
+            postal_code = request_body["postal_code"],
+            phone = request_body["phone"],
+            register_at = date.today()
+            )
+        db_add(new_customer)
+        return make_response(new_customer.to_dict(), 201)
+    
+    else:
+        return make_response(jsonify({"message": "http request method not permitted"}), 400)
+
+@customers_bp.route("/<customer_id>", methods=["GET", "DELETE", "PUT"])
+def handle_one_customer(customer_id):
+    """Defined an endpoint to handle all specified customer http request methods"""
+    customer = is_class_type_at_id_found(customer_id, Customer, "Customer")
+
+    if request.method == "GET":
+        return make_response(jsonify(customer.to_dict()), 200)
+
+    elif request.method == "DELETE":
+        if not customer.rentals:
+            db_delete(customer)
+            return make_response(jsonify({"id": int(customer_id)}), 200)
+        else:
+            active_rentals = Rental.query.filter_by(customer_id=f"{customer.id}")
+            for rental in active_rentals:
+                db.session.delete(rental)
+            db_delete(customer)
+            return make_response("", 200)
```

Comment:
Is the assumption here that the customer did not return their currently rented videos?  The variable name `active_rentals` implies that these are currently checked out, but the query will bring up all rentals past and present.  To delete only the rentals where `status==checked_out`, Rental.query.filter(...).all() makes it possible to input multiple selection criteria (ie, customer_id & status) (https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query)